include/ginkgo/core/distributed/matrix.hpp

```diff
@@ -242,6 +252,7 @@ class Matrix
     friend class EnableDistributedPolymorphicObject<Matrix, LinOp>;
     friend class Matrix<next_precision<ValueType>, LocalIndexType,
                         GlobalIndexType>;
+    friend class multigrid::Pgm<ValueType, LocalIndexType>;
```

Comment:
Yes, because previous approach require additional features class with many functions. Because we only use these in PGM currently and there are some future changes, we use friend to put these still internal. When we have more than one class for that, we definitely should think about how to extract them to feature class or alwasy use dispatch.